webapp/packages.py

```diff
@@ -23,6 +24,23 @@ class PackagesAPI:
     def __init__(self, cache):
         self.cache = cache
 
+    def find_packages_by_regex(self, regex):
+        """Returns list of packages matching a provided regex."""
+        if not regex.startswith('^'):
+            regex = '^' + regex
+
+        if not regex.endswith('$'):
+            regex = regex + '$'
+
+        filtered_pkgs = []
+        for (name_id, evr_id, arch_id) in self.cache.nevra2pkgid:
+            pkg_id = self.cache.nevra2pkgid[(name_id, evr_id, arch_id)]
+            package_list = self._get_built_binary_packages(pkg_id)
```

Comment:
this is probably overly complicated and doesn't work for source packages

would something like this work?
```
for (name_id, evr_id, arch_id) in self.cache.nevra2pkgid:
    # get name, epoch, version, release, arch
    # ...
    variants = ["%s-%s:%s-%s.%s" % (name, epoch, version, release, arch)]
    if epoch == 0:
        variants.append("%s-%s-%s.%s" % (name, version, release, arch))
    for item in variants:
        if re.match(regex, item):
            filtered_pkgs.append(item)
```

Reply:
I found a little bit another way how to do it
```
        regex = packages[0]
        name, epoch, ver, rel, arch = utils.split_packagename(regex)
        if name and epoch and ver and rel and arch:
            # If there is no regular expression return all packages what we get in api request
            return packages
```
Looks like it works fine